OOInteraction/src/commands/CAddNodeToView.cpp

```diff
@@ -35,28 +35,35 @@ CAddNodeToView::CAddNodeToView()
 {
 }
 
-bool CAddNodeToView::canInterpret(Visualization::Item *source, Visualization::Item *target,
-	const QStringList &commandTokens, const std::unique_ptr<Visualization::Cursor> &cursor)
+bool CAddNodeToView::canInterpret(Visualization::Item* source, Visualization::Item* target,
+	const QStringList& commandTokens, const std::unique_ptr<Visualization::Cursor>& cursor)
 {
-	 return Interaction::CommandWithDefaultArguments::canInterpret(source, target, commandTokens, cursor)
-			 && target->hasNode() && dynamic_cast<OOModel::Method*>(target->node());
+	bool canInterpret = CommandWithDefaultArguments::canInterpret(source, target, commandTokens, cursor);
+	//The first parent with a node should be a declaration (these can be added to the view here)
+	while (source && !source->hasNode())
+		source = source->parent();
+	if (!source)
+		return false;
+	else
+		return canInterpret && dynamic_cast<OOModel::Declaration*>(source->node());
```

Comment:
Here and everywhere else, whenever you deal with objects that derive from either Item or Node, use DCast<Type>(object) instead of dynamic_cast<Type>(object)